lib/Remote/RemoteKeyboard.php

```diff
@@ -37,7 +37,7 @@ public function sendKeys($keys)
     {
         if ($this->isW3cCompliant) {
             $activeElement = $this->driver->switchTo()->activeElement();
-            $activeElement->sendKeys($keys);
+            $activeElement->sendKeys(WebDriverKeys::encode($keys, true));
```

Comment:
`RemoteWebElement::sendKeys()` is also encoding the values which can lead to a double-encoded value. Strictlly speaking, this doesn't matter because the purpose of `encode` function is largely that of normalisation. The encoded value can be safely re-encoded without any ill effect.

That said, if we were to make changes to `encode` in the future such that it can no longer be called on an already-encoded value, then `sendKeys` would be sending a double-encoded value and this may be difficult to detect.

I'd suggest that we either drop the `encode()` call here entirely (and in the OSS variant too if possible), or we add relevant unit tests to ensure that a value such as the following does not break:

```
$this->driver->getKeyboard()->sendKeys([
    [WebDriverKeys::SHIFT . 'h'],
    'elp',
]);
```

Reply:
Nice catch, the call is really redundant here and should not be there.

The encode call needs to stay in the OSS version though, because it is not send via `sendKeys()` and it needs to be converted to array of chars (which is the way JsonWire protocol accepts text).